app/service/rest.py

```diff
@@ -1322,3 +1328,72 @@ def get_service_join_request(request_id: uuid.UUID):
         raise InvalidRequest(message=f"Service join request with ID {request_id} not found.", status_code=404)
 
     return jsonify(service_join_request.serialize()), 200
+
+
+@service_blueprint.route("/update-service-join-request-status/<uuid:request_id>", methods=["POST"])
+def update_service_join_request(request_id: uuid.UUID):
+    data = request.get_json()
+
+    validate(data, service_join_request_update_schema)
+
+    status = data["status"]
+    status_changed_by_id = data["status_changed_by_id"]
+    reason = data.get("reason", None)
+
+    updated_request = dao_update_service_join_request(request_id, status, status_changed_by_id, reason)
+
+    if updated_request is None:
+        return jsonify({"message": "Service join request not found"}), 404
+
+    if status == SERVICE_JOIN_REQUEST_APPROVED:
+        permissions = data.get("permissions", None)
+
+        if permissions:
+            permissions = [
+                Permission(service_id=updated_request.service_id, user_id=updated_request.requester_id, permission=p)
+                for p in permissions
+            ]
+
+        requester_user = get_user_by_id(updated_request.requester_id)
+        approver_user = get_user_by_id(updated_request.status_changed_by_id)
+        service = dao_fetch_service_by_id(updated_request.service_id)
+
+        dao_add_user_to_service(service, requester_user, permissions)
+
+        send_service_join_request_decision_email(
+            requester_email_address=requester_user.email_address,
+            template_id=current_app.config["SERVICE_JOIN_REQUEST_APPROVED_TEMPLATE_ID"],
+            personalisation=create_personalisation(requester_user.name, approver_user.name, service.name, service.id),
+        )
+
+        dao_cancel_pending_service_join_requests(requester_user.id, approver_user.id, service.id)
+
+    return jsonify(updated_request.serialize()), 200
+
+
+def create_personalisation(requester_name: str, approver_name: str, service_name: str, service_id: uuid):
+    admin_base_url = current_app.config["ADMIN_BASE_URL"]
+    return {
+        "requester_name": requester_name,
+        "approver_name": approver_name,
+        "service_name": service_name,
+        "dashboard_url": f"{admin_base_url}/services/{service_id}",
+    }
+
+
+def send_service_join_request_decision_email(requester_email_address: str, template_id: uuid, personalisation: dict):
+    notify_service = dao_fetch_service_by_id(current_app.config["NOTIFY_SERVICE_ID"])
+
+    saved_notification = persist_notification(
+        template_id=template_id,
+        template_version=1,
```

Comment:
You'll need to get the current version of the template to send, which means that we need to get the template out of the database. There's an example of where we've done that in this endpoint you could copy https://github.com/alphagov/notifications-api/blob/98ade3b931eb4d2f4aa232802aa134ee48f18b59/app/organisation/rest.py#L332